### DISCUSSION TITLE
Next13 Head: Sharing data between page.tsx and head.tsx?

### DISCUSSION THREAD

**Discussion by williamlmao** [vercel/next.js#43095] [Help] [answered]
[The docs ](https://beta.nextjs.org/docs/api-reference/file-conventions/head) currently show an example where you fetch data inside of head. Most of the time, you are probably already fetching `post` within `page.tsx`. Is there any way to share the data between the two files so you don't have to fetch the same data twice?

```
async function getPost(slug) {
  const res = await fetch('...');
  return res.json();
}

export default async function Head({ params }) {
  const post = await getPost(params.slug);

  return (
    <>
      <title>{post.title}</title>
    </>
  )
}
```

**Comment by paulyoung5** [comment, marked as answer]
I don't think you can pass data as such but check out the new `cache` function (exported from `react`):

https://beta.nextjs.org/docs/data-fetching/caching#per-request-caching

I think if you wrap your fetch function with that, you can call the same function in both places and Next should dedupe the requests, so the request will only be made once 👍

**Reply by williamlmao** [reply]
Oooh, yeah this makes it easy. Thanks Paul!
